internal/server/spanner/datasource.go

```diff
@@ -68,15 +68,31 @@ func (sds *SpannerDataSource) Node(ctx context.Context, req *v3.NodeRequest) (*v
 
 // Observation retrieves observation data from Spanner.
 func (sds *SpannerDataSource) Observation(ctx context.Context, req *v3.ObservationRequest) (*v3.ObservationResponse, error) {
-	// Only variable and entity dcids are supported for now.
-	// TODO: Add support for expressions.
-	variables, entities := req.Variable.Dcids, req.Entity.Dcids
+	variables, entities, entityExpr := req.Variable.Dcids, req.Entity.Dcids, req.Entity.Expression
 	date := req.Date
 
-	observations, err := sds.client.GetObservations(ctx, variables, entities)
-	if err != nil {
-		return nil, fmt.Errorf("error getting observations: %v", err)
+	var observations []*Observation
+
+	if entityExpr != "" {
+		containedInPlace, err := v2.ParseContainedInPlace(entityExpr)
+		if err != nil {
+			return nil, fmt.Errorf("error getting observations (contained in): %v", err)
+		}
+		observations, err = sds.client.GetObservationsContainedInPlace(ctx, variables, containedInPlace)
```

Comment:
Were there any performance concerns with fetching the full series for all the children places?

Reply:
For the golden tests it was fine. For something like "Count_Person" and "country/USA<-containedInPlace+{typeOf:State}" it was noticeable.